### PR TITLE
Use discover peers function in p2p - Closes#979

### LINKED ISSUE #979: Use discoverPeers function in p2p for discovery
### Description

Use `discoverPeers` function in p2p to do peer discovery.

[peer_discovery.ts](https://github.com/LiskHQ/lisk-elements/blob/add_p2p_experiment/packages/lisk-p2p/src/peer_discovery.ts)

### PR DESCRIPTION
**Pull request by ishantiw**
### Description

Use discovery in P2P to find new peers. It should take initial peers on which the discovery algo will run and then it will return all the relevant peers based on the algo and add it to `newPeers` list.

* The PR resolves #979 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Review by shuse2 — APPROVED** [review]

File: packages/lisk-p2p/src/p2p.ts
Comment (shuse2) [inline packages/lisk-p2p/src/p2p.ts]: This should be `forEach` if it's not mapping

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]

**Comment by ishantiw** [comment]
Rebased and ready for review, waiting for your approval @jondubois

**Review by jondubois — APPROVED** [review]
Will need to resolve merge conflicts in https://github.com/LiskHQ/lisk-elements/pull/993/files (if any).

**Pull request merged**
